app/models/under_rep_well_comments_to_broadcast.rb

```diff
@@ -0,0 +1,135 @@
+  # with the `UNDER_REPRESENTED` key.
+  #
+  # @return [Array<Request>] an array of requests that contain under-represented metadata.
+  def request_with_under_represented_wells
+    submissions.flat_map(&:requests).select do |r|
+      r.poly_metadata.any? { |pol| pol.key == UNDER_REPRESENTED_KEY }
+    end
+  end
+
+  ##
+  # Builds all `UnderRepWellComment` objects for requests within the batch
+  # that have under-represented wells. The resulting list aggregates comments
+  # across multiple requests and associated assets.
+  #
+  # @return [Array<UnderRepWellComment>] a flat list of comment objects.
+  #
+  def under_represented_well_comments
+    request_with_under_represented_wells.flat_map do |request|
+      build_comments_for_request(request)
+    end.compact
+  end
+
+  # Entry point used by CommentIO to retrieve the comments to serialize.
+  #
+  # @return [Array<UnderRepWellComment>] comments representing under-represented wells.
+  def comments
+    under_represented_well_comments || []
+  end
+
+  private
+
+  ##
+  # Constructs `UnderRepWellComment` objects for a single request,
+  # filtering only poly_metadata that correspond to under-represented wells.
+  #
+  # @param request [Request] the request to process
+  # @return [Array<UnderRepWellComment>] comments built for the given request
+  #
+  def build_comments_for_request(request)
+    under_represented_poly_metadata(request).flat_map do |poly_meta|
+      build_comments_for_poly_meta(request, poly_meta)
+    end
+  end
+
+  ##
+  # Returns only those `PolyMetadatum` records whose key matches
+  # the `UNDER_REPRESENTED_KEY` constant.
+  #
+  # @param request [Request] the request whose poly_metadata to inspect
+  # @return [Array<PolyMetadatum>] filtered poly_metadata records
+  #
+  def under_represented_poly_metadata(request)
+    request.poly_metadata.select { |meta| meta.key == UNDER_REPRESENTED_KEY }
+  end
+
+  ##
+  # Builds comments for a specific poly_metadatum by iterating through
+  # all target aliquots and matching them to lane aliquots by tag map ID.
+  #
+  # The starting point is the request that holds the poly_metadatum for the
+  # under-represented well. This request is typically a LibraryRequest, whose
+  # asset corresponds to the under-represented well.
+  #
+  # From there, we traverse to the Lane asset via `request.asset.descendants`.
+  # The Lane provides access to the `batch_request` (through `lane.source_request`),
+  # which contains the lane position information used in the generated comment.
+  #
+  # @param request [Request] the request that owns the poly_metadatum
+  # @param poly_meta [PolyMetadatum] the metadata record for under-represented wells
+  # @return [Array<UnderRepWellComment>] list of constructed comment objects
+
+  def build_comments_for_poly_meta(request, poly_meta)
+    request.target_asset.aliquots.filter_map do |aliquot|
+      lane = request.asset.descendants.last
```

Comment:
Another possibly fragile point if used in more recent, more complicated pipelines - not confident whether the `descendants` links would all be intact. OK for now.